### PR TITLE
Convert to API

### PR DESCRIPTION
**Pull request by sevenreup**
- [x] get range download from request
- [x] restructure file save paths (add range modifiers and data)
- [x] add daily download
- [ ] update readme

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+RAW_PDF_PATH=
+RAW_CSV_PATH=
+ERROR_FILE_PATH=
+CLEANED_JSON_PATH=
+CLEANED_CSV_PATH=
+QUEUE_SIZE=
+WORKER_NUM=
+PDFTABLES_API_KEY=
+DB_CONNECTION_STRING=
+PORT=
```
Comment (liwoo) [inline .env.example]: What's this port for?

File: api/clean/cleaner.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package clean
```
Comment (liwoo) [inline api/clean/cleaner.go]: Should be in the handlers package of http, I.e. http/handlers/cleaner

File: api/clean/cleaner.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+	fmt.Println("Saving to Database from... ", s.CleanedJsonPath)
+	p := pool.NewPool(s.QueueSize, s.WorkerNum)
+
+	var pgconn *pgdriver.Connector = pgdriver.NewConnector(pgdriver.WithDSN(s.DBConnectionString))
+	psdb := sql.OpenDB(pgconn)
+	db := bun.NewDB(psdb, pgdialect.New())
+
+	p.Start()
+
+	dbCleaner.CleanDatabase(db)
```
Comment (liwoo) [inline api/clean/cleaner.go]: Where is this coming from. All dependencies must be explicitly defined and passed on by parents

File: main.go [hunk 4]
```diff
@@ -1,56 +1,36 @@
...
-	pdfStartNo, pdfStartNoError := strconv.Atoi(envs["PDF_START_NO"])
-	pdfEndNo, pdfEndNoError := strconv.Atoi(envs["PDF_END_NO"])
+	app.Router.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("Welcome to MSE Scrapper"))
+	})
 
-	if queueSizeError != nil || workerNumError != nil || pdfStartNoError != nil || pdfEndNoError != nil {
-		log.Panicf("Cannot convert Queue Size or Worker Nums to Int")
-	}
+	app.Router.Route("/scrapper", func(r chi.Router) {
```
Comment (liwoo) [inline main.go]: Should have an api prefix

File: pkg/server/server.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+type HandleWithBody func(body any)
+
+type Server struct {
+	Conf   conf.Conf
+	Router *chi.Mux
+}
+
+func Build() Server {
```
Comment (liwoo) [inline pkg/server/server.go:21]: Can we parse vars from os env?

**Review by liwoo — CHANGES_REQUESTED** [review]
Would be nice if we could have some tests... but I guess we can scaffold those together

File: .env.example (hunk 1 above)
Comment (sevenreup) [reply .env.example]: The port that the server runs on

File: api/clean/cleaner.go (hunk 2 above)
Comment (sevenreup) [reply api/clean/cleaner.go]: this was for cleaning up the DB

File: api/clean/cleaner.go (hunk 3 above)
Comment (sevenreup) [reply api/clean/cleaner.go]: moved

File: pkg/server/server.go (hunk 5 above)
Comment (sevenreup) [reply pkg/server/server.go:21]: Yes we can parse os env since we are getting it from `os.Getenv`

File: .env.example (hunk 1 above)
Comment (liwoo) [reply .env.example]: let's call it `SERVER_PORT`

File: api/clean/cleaner.go (hunk 2 above)
Comment (liwoo) [reply api/clean/cleaner.go]: but these are HTTP handlers no?

File: api/scrapper/scrapper.go [hunk 6]
```diff
@@ -0,0 +1,162 @@
+package scrapper
```
Comment (liwoo) [inline api/scrapper/scrapper.go]: same thing with these, they should be in `http/handlers`

**Review by liwoo — CHANGES_REQUESTED** [review]
File structure

http
   --> handlers
data
  --> actual database connectors, sql etc
services
  --> containers for logic for cleaner/downloader etc
models
   --> anything that goes in the database
shared
    -> any shared DTOs (api responses/requests)

**Review by liwoo — APPROVED** [review]

**Pull request merged**